### DISCUSSION TITLE
Replace RSS by Atom

### DISCUSSION THREAD

**Discussion by sjehuda** [mastodon/mastodon#29892] [General]
### `TL:DR`

RSS is inherently broken

Stop using that and move to the proper syndication format that is actually properly defined and easy to parse. All the syndication clients support both so there is no excuse to continue to generate broken RSS feeds.

https://datatracker.ietf.org/doc/html/rfc4287

"Since 2005"

---

Greeting!

Here is a good post about Atom and RSS (by Chris)

> [chrismorgan](https://news.ycombinator.com/user?id=chrismorgan) [on Feb 17, 2021](https://news.ycombinator.com/item?id=26169162) | [parent](https://news.ycombinator.com/item?id=26168493) | [context](https://news.ycombinator.com/context?id=26169162) | [favorite](https://news.ycombinator.com/fave?id=26169162&auth=c47a5cea67e54c67e902800088f92d2bcaa097c9) | on: [Kill the Newsletter – Convert email newsletters in...](https://news.ycombinator.com/item?id=26167248)
>
> Atom is technically substantially superior to RSS. RSS is riddled with ambiguities which lead to inconsistency of interpretation, and being outright unable to express some useful things.
My favourite is the format of all content: is it HTML or text? RSS doesn’t care, and leaves you to guess. So people added a namespace to fix that up, and you end up with what I think is normally spelled <content:encoded> to replace <description> or whatever it is—but there’s no equivalent for other content fields like titles. And because you can’t trust that tool will use <content:encoded>, now you need both. (And when you get to podcasts, the theme of the day is having the same field expressed in four or five different namespaces for different tools, in some cases for things that Atom has but RSS lacks.)
>
> I have an article on my blog entitled `<_>::v::<_>`. I expect that a few RSS readers would mangle it if I served an RSS feed, but I serve an Atom feed, so I can express the title correctly:
> ```
> <title type="html">&lt;code&gt;&amp;lt;_&amp;gt;::v::&amp;lt;_&amp;gt;&lt;&#x2F;code&gt;</title>
> ```
> (Bet you didn’t anticipate the `<code>` wrapping of that title! Now try that in RSS!) Anyway, NewsBlur got it right, and if any other client gets it wrong I can point to it and say that it is unequivocally a bug, whereas with RSS… eh, there’s enough ambiguity to drive a spaceship through, so what does “bug” even mean?
I hate the way that feeds are so commonly generically called “RSS”, because it gives mindshare to the bad and inferior product.

---

@edhelas has concisely conveyed to me that:

> RSS is inherently broken
> 
> Stop using that and move to the proper syndication format that is actually properly defined and easy to parse. All the syndication clients support both so there is no excuse to continue to generate broken RSS feeds.
> 
> https://datatracker.ietf.org/doc/html/rfc4287
> "Since 2005"

---

### There is no reason you should ever generate RSS feeds.

**Comment by markcellus** [comment]
I agree that RSS may not have the most robust features. I doubt anyone will argue with you on that. But, regardless of whether people think RSS is "broken", telling RSS developers not to build RSS is naive. Anyone who's been working on RSS for as long as many of us have will tell you that RSS 2.0 ensures the widest compatibility between RSS readers. There are actually RSS readers that have been around for ages that don't parse Atom feeds correctly. And, going around forcing all of those RSS readers to switch to Atom is impossible. So, at best, this request should be to add Atom feeds as _an option_, but RSS 2.0 (at least) should still be supported.

---
As a side note: with all due respect, the tone of your messaging (in this post and others) comes off condescending and arrogant. You would help yourself by improving the way you're asking for what you need. We're all open source developers, many working on the apps and services you're using for free, and are only trying to do our best. The least we can do when requesting new features is to try to show a little more empathy.

**Reply by sjehuda** [reply]
> As a side note: with all due respect, the tone of your messaging (in this post and others) comes off condescending and arrogant. You would help yourself by improving the way you're asking for what you need. We're all open source developers, many working on the apps and services you're using for free, and are only trying to do our best. The least we can do when requesting new features is to try to show a little more empathy.

Noted!
Thank you very much for the comment!

Please pardon.
I did not mean to make such an impression.

Albeit I do not use Mastodon _yet_ (I am still in Email and XMPP), I greately appreciate your work, Mastodon, and other ActivityPub projects.

**Reply by sjehuda** [reply]
> And, going around forcing all of those RSS readers to switch to Atom is impossible. So, at best, this request should be to add Atom feeds as an option, but RSS 2.0 (at least) should still be supported.

Which Feed readers do not provide support for Atom Syndication?

**Reply by markcellus** [reply]
> Which Feed readers do not provide support for Atom Syndication?|

There's no possible way to know that and always have an updated list, which is my point. You don't know every possible reader out there that only supports RSS 1.0 or 2.0 now or in the future, so pushing to stop using RSS is shortsighted and ignores this fact.

RSS 1.0/2.0 isn't going to just go away because we say so. It's a protocol that existed before Atom, and sites and apps will continue to use it, regardless of what we think.

To be clear, I'm not saying Mastodon shouldn't provide Atom feeds. I'm saying Mastodon shouldn't stop supporting RSS 1.0/2.0 feeds, because it would limit their use in RSS apps and services.

**Reply by sjehuda** [reply]
Is it acceptable for me to provide or help in providing a patch for Atom?
